zerver/lib/user_topics.py

```diff
@@ -123,17 +121,16 @@ def set_user_topic_visibility_policy_in_database(
     database_changed: bool = True
 
     if visibility_policy == UserTopic.VisibilityPolicy.INHERIT:
-        try:
-            # Will throw UserTopic.DoesNotExist if the user doesn't
-            # already have a visibility policy for this topic.
-            UserTopic.objects.get(
-                user_profile=user_profile,
-                stream_id=stream_id,
-                topic_name__iexact=topic_name,
-            ).delete()
-            return database_changed
-        except UserTopic.DoesNotExist:
-            raise JsonableError(_("Nothing to be done"))
+        deleted = UserTopic.objects.filter(
+            user_profile=user_profile,
+            stream_id=stream_id,
+            topic_name__iexact=topic_name,
+        ).delete()
+        if deleted[0] == 0:
```

Comment:
I prefer `(num_deleted, ignored) = UserTopic.objects...` as a pattern if something returns a tuple.

Reply:
I will update this.

I just [followed](https://github.com/zulip/zulip/blob/main/zilencer/views.py#L200) the existing pattern in the codebase.

Reply:
This pattern is definitely nicer, so it wouldn't be a bad idea to update existing code using that pattern in a little follow-up PR.

Reply:
@timabbott 

Here's the follow-up PR #24737 

ps: I have verified that there is no more existing code in the codebase that doesn't follow this pattern.